central/reports/service/v2/convert.go

```diff
@@ -376,38 +380,64 @@ func ConvertProtoNotifierSnapshotToV2(notifierSnapshot *storage.NotifierSnapshot
 }
 
 // convertPrototoV2ReportSnapshot converts storage.ReportSnapshot to apiV2.ReportSnapshot
-func convertProtoReportSnapshotstoV2(snapshots []*storage.ReportSnapshot) []*apiV2.ReportSnapshot {
+func (s *serviceImpl) convertProtoReportSnapshotstoV2(snapshots []*storage.ReportSnapshot) ([]*apiV2.ReportSnapshot, error) {
 	if snapshots == nil {
-		return nil
+		return nil, nil
 	}
-	res := make([]*apiV2.ReportSnapshot, 0, len(snapshots))
+	blobNames, err := s.getExistingBlobNames(snapshots)
```

Comment:
This seems expensive. Why not search?

Reply:
Same question as [here](https://github.com/stackrox/stackrox/pull/7410#discussion_r1293838183)